Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -461,7 +470,7 @@ function Get-AzRmResourceGroup
 
         $proxyUri = Get-ProxyUri $uri
 
-        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $uri))
+        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($uri -like "*$($proxyUri.Host)*"))
```

Comment:
Apparently not. I think I raised a PR to add proxy support for the Azure SQL task but for some reason I never merged it. That PR is still open.